### DISCUSSION TITLE
Route prefix for layers

### DISCUSSION THREAD

**Discussion by Kronhyx** [nuxt/nuxt#28123] [Questions] [answered]
<img width="328" alt="image" src="https://github.com/nuxt/nuxt/assets/16559276/8ade2245-2521-4e72-913a-d5398fd36697">

is there any way to prefix layer routes ?

each layers has their own page router system, but in order to avoid collision on names i would like to add a route prefix for each layer... is this possible ?

**Comment by mavyfaby** [comment]
same question

**Comment by spyesx** [comment]
The only way I found is to create a directory in the layer's pages dir.

```
layers 
  my-layer
    pages
       my-namespace
```

However, it would be nice to use something like `baseURL` for `layerBaseUrl`.

**Reply by mavyfaby** [reply]
for temporary workaround, yes

**Comment by zangetsu02** [comment]
any updates about this?

**Comment by L1nuxNB** [comment]
would be great for server routes as well...

**Comment by YouneselBarnoussi** [comment]
Would be nice to have this functionality added

**Comment by Kronhyx** [comment]
@danielroe ....

**Comment by cernymatej** [comment]
Hello, you can already do this via a hook. In each of your layers, use this:

```ts
import { createResolver } from 'nuxt/kit'
import { joinURL } from 'ufo'

const { resolve } = createResolver(import.meta.url)

// here you resolve a path to the pages/ directory relative to the layer's nuxt config
const layerPagesDir = resolve('./app/pages')

export default defineNuxtConfig({
  hooks: {
    'pages:extend'(pages) {
      for (const page of pages) {
        if (page.file?.startsWith(layerPagesDir)) {
          // here you can transform the page path however you like:
          page.path = joinURL('/admin', page.path)
        }
      }
    },
  },
})
```

**Comment by Kronhyx** [comment, marked as answer]
Hi everyone, has been solve for nuxt 4.3.0 by @danielroe using

https://nuxt.com/blog/v4-3#%EF%B8%8F-route-rule-layouts

```ts
export default defineNuxtConfig({
  routeRules: {
    '/admin/**': { appLayout: 'admin' },
    '/dashboard/**': { appLayout: 'dashboard' },
    '/auth/**': { appLayout: 'minimal' }
  }
})
```

cc: @mavyfaby @spyesx @zangetsu02 @L1nuxNB @YouneselBarnoussi @cernymatej

**Reply by spyesx** [reply]
Thx for the update ! Gonna try.

**Reply by zangetsu02** [reply]
yo tysm!

**Reply by mavyfaby** [reply]
@Kronhyx how does this solve the route prefix? How can the layout prefix routes?
